projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -759,16 +772,151 @@ function wpcom_add_new_page_check( $post_id, $post ) {
+	$headstart_about_pages = array_filter(
+		$annotation['content'],
+		function ( $page ) {
+			if ( 'page' !== $page['post_type'] ) {
+				return false;
+			}
+
+			if ( 'about' === $page['post_name'] || false !== strpos( $page['post_title'], 'About' ) ) {
+				return true;
+			}
+		}
+	);
+
+	// Return null if there are no 'About' pages.
+	if ( empty( $headstart_about_pages ) ) {
+		return null;
+	}
+
+	// Get the hs_old_ids for the 'About' pages. We'll use these to find any published pages with the same hs_old_id.
+	$headstart_about_page_hs_old_ids = array_map(
+		function ( $about_page ) {
+			return $about_page['hs_old_id'];
+		},
+		$headstart_about_pages
+	);
+
+	// Return null if there aren't any hs_old_ids to check.
+	if ( empty( $headstart_about_page_hs_old_ids ) ) {
+		return null;
+	}
+
+	$filters     = array(
+		'post_type'    => 'page',
+		'numberposts'  => 1,
+		'orderby'      => 'meta_value',
+		'meta_key'     => '_hs_old_id',
+		'meta_compare' => 'IN',
+		'meta_value'   => $headstart_about_page_hs_old_ids,
+	);
+	$about_pages = get_posts( $filters );
+
+	// Return null if we couldn't find any published pages matching the hs_old_ids.
+	if ( empty( $about_pages ) ) {
+		return null;
+	}
+
+	// Cache to hs_old_ids so we can validate the post being updated later.
+	wp_cache_set( "headstart_about_page_hs_old_ids_{$blog_id}", $headstart_about_page_hs_old_ids, 'wpcom' );
+
+	// Cache the about page id so we don't have to look it up from scratch every time.
+	wp_cache_set( $about_page_id_cache_key, $about_pages[0]->ID, 'wpcom' );
+
+	// Return the id of the first About page.
+	return $about_pages[0]->ID;
+}
+
+/**
+ * Determine `update_about_page` task visibility. The task is visible if there is an 'About' page on the site.
+ *
+ * @return bool True if we should show the task, false otherwise.
+ */
+function wpcom_is_update_about_page_task_visible() {
+	return wpcom_get_site_about_page_id() !== null;
+}
+
+/**
+ * When a page is updated, check to see if it's the About page and mark the task complete accordingly.
+ *
+ * @param int     $post_id The ID of the post being updated.
+ * @param WP_Post $post The post object.
+ */
+function wpcom_update_about_page_check( $post_id, $post ) {
+	// Ensure that Headstart posts don't mark this as complete
+	if ( defined( 'HEADSTART' ) && HEADSTART ) {
+		return;
+	}
+
+	// We only care about pages, ignore other post types.
+	if ( $post->post_type !== 'page' ) {
+		return;
+	}
+
+	// We only care about published pages. Pages added via the API are not published by default.
+	if ( $post->post_status !== 'publish' ) {
+		return;
+	}
+
+	// Don't do anything if the task is already complete.
+	if ( wpcom_is_task_option_completed( array( 'id' => 'update_about_page' ) ) ) {
+		return;
+	}
+
+	// If the page is not the previously located About page, ignore it.
+	$blog_id       = get_current_blog_id();
+	$about_page_id = wp_cache_get( "wpcom_get_site_about_page_id_{$blog_id}", 'wpcom' );
```

Comment:
We shouldn't use the cache here -- we should call `wpcom_get_site_about_page_id()` so we have one code block in charge of the cache retrieval and updates, and that allows us to compute the value if it hasn't been cached. Given that, we should probably change this logic slightly.
```suggestion
	$about_page_id = wpcom_get_site_about_page_id();
	if ( null === $about_page_id || $post->ID !== $about_page_id ) {
		return;
	}
```